src/components/product/product.service.ts

```diff
@@ -370,105 +345,102 @@ export class ProductService {
       throw new NotFoundException('Could not find product', 'product.id');
     }
 
-    const props = parsePropList(result.propList);
-    const securedProperties = parseSecuredProperties(
-      props,
-      result.permList,
-      this.securedProperties
-    );
-    const baseNodeProps = parseBaseNodeProperties(result.node);
+    const {
+      produces,
+      scriptureReferencesOverride,
+      isOverriding,
+      ...rest
+    } = parseSecuredProperties(result.propList, result.permList, {
+      mediums: true,
+      purposes: true,
+      methodology: true,
+      scriptureReferences: true,
+      scriptureReferencesOverride: true,
+      produces: true,
+      isOverriding: true,
+    });
 
-    const produces = await this.db
+    const pr = await this.db
       .query()
       .match([
         node('product', 'Product', { id, active: true }),
         relation('out', 'produces', { active: true }),
         node('p', 'Producible', { active: true }),
       ])
       .return('p')
+      .asResult<{ p: Node<BaseNode> }>()
       .first();
 
-    if (!produces) {
-      const scriptureReferences = await this.listScriptureReferences(
-        baseNodeProps.id,
-        'Product',
-        session
-      );
+    const scriptureReferencesValue = await this.scriptureRefService.list(
+      id,
+      session,
+      { isOverriding: pr ? true : false }
+    );
+
+    if (!pr) {
       return {
-        ...baseNodeProps,
-        ...securedProperties,
+        ...parseBaseNodeProperties(result.node),
+        ...rest,
         scriptureReferences: {
-          canRead: result.canScriptureReferencesRead,
-          canEdit: result.canScriptureReferencesEdit,
-          value: scriptureReferences,
+          ...rest.scriptureReferences,
+          value: scriptureReferencesValue,
         },
         mediums: {
-          ...securedProperties.mediums,
-          value: securedProperties.mediums.value ?? [],
+          ...rest.mediums,
+          value: rest.mediums.value ?? [],
         },
         purposes: {
-          ...securedProperties.purposes,
-          value: securedProperties.purposes.value ?? [],
+          ...rest.purposes,
+          value: rest.purposes.value ?? [],
         },
       };
     }
 
-    const scriptureReferencesOverride = await this.listScriptureReferences(
-      baseNodeProps.id,
-      'Product',
-      session,
-      { isOverride: true }
-    );
-
-    const typeName = difference(produces.p.labels, [
-      'Producible',
-      'BaseNode',
-    ])[0];
+    const typeName = difference(pr.p.labels, ['Producible', 'BaseNode'])[0];
```

Comment:
```suggestion
    const typeName = difference(pr.p.labels as ProducibleType[], ['Producible', 'BaseNode'])[0];
```
This should prevent the need to typecast to any below.